rle/rleplus.go

```diff
@@ -36,18 +37,18 @@ func (rle *RLE) RunIterator() (RunIterator, error) {
 		if err != nil {
 			return nil, xerrors.Errorf("decoding RLE: %w", err)
 		}
+		var length uint64
 		for source.HasNext() {
 			r, err := source.NextRun()
 			if err != nil {
 				return nil, xerrors.Errorf("reading run: %w", err)
 			}
+			if math.MaxUint64-r.Len < length {
+				return nil, xerrors.New("RLE+ overflows")
```

Comment:
We should set `rle.runs = nil` just to be extra safe.

Reply:
This way we don't leave inconsistent state.

Reply:
I'm using a temporary variable now.